tests/test_penalty.py

```diff
@@ -1127,3 +1129,56 @@ def test_penalty_track_markers_with_nan(penalty_origin, value):
         np.array(res),
         expected,
     )
+
+
+@pytest.mark.parametrize("node", [Node.ALL, Node.INTERMEDIATES, Node.START, Node.MID, Node.PENULTIMATE, Node.END,
+                                  Node.TRANSITION])
+@pytest.mark.parametrize("ns", [10, 11])
+def test__get_node(node, ns):
+
+    NLP = nlp()
+    NLP.ns = ns
+    NLP.X = np.linspace(0, -10, ns+1)
+    NLP.U = np.linspace(10, 19, ns)
+
+    pn = []
+    penalty = PenaltyOption(pn)
+    penalty.node = node
+
+    if node == Node.MID and ns % 2 != 0:
+        with pytest.raises(ValueError, match="Number of shooting points must be even to use MID"):
+            t, x, u = PenaltyFunctionAbstract._get_node(NLP, penalty)
```

Comment:
Not sure, but I think a return is missing here.. otherwise the function will continue while there is nothing more to test if get node fails?

Reply:
That's the point I think. Since it will fail I don't want to test anything more and that case.